src/portals/Screenshot.cpp

```diff
@@ -0,0 +1,123 @@
+#include "Screenshot.hpp"
+#include "../core/PortalManager.hpp"
+#include "../helpers/Log.hpp"
+#include "../helpers/MiscFunctions.hpp"
+#include <regex>
+
+CScreenshotPortal::CScreenshotPortal() {
+    m_pObject = sdbus::createObject(*g_pPortalManager->getConnection(), OBJECT_PATH);
+
+    m_pObject->registerMethod(INTERFACE_NAME, "Screenshot", "ossa{sv}", "ua{sv}", [&](sdbus::MethodCall c) { onScreenshot(c); });
+    m_pObject->registerMethod(INTERFACE_NAME, "PickColor", "ossa{sv}", "ua{sv}", [&](sdbus::MethodCall c) { onPickColor(c); });
+
+    m_pObject->registerProperty(INTERFACE_NAME, "version", "u", [](sdbus::PropertyGetReply& reply) -> void { reply << (uint32_t)(2); });
+
+    m_pObject->finishRegistration();
+
+    Debug::log(LOG, "[screenshot] init successful");
+}
+
+void CScreenshotPortal::onScreenshot(sdbus::MethodCall& call) {
+    sdbus::ObjectPath requestHandle;
+    call >> requestHandle;
+
+    std::string appID;
+    call >> appID;
+
+    std::string parentWindow;
+    call >> parentWindow;
+
+    std::unordered_map<std::string, sdbus::Variant> options;
+    call >> options;
+
+    Debug::log(LOG, "[screenshot] New screenshot request:");
+    Debug::log(LOG, "[screenshot]  | {}", requestHandle.c_str());
+    Debug::log(LOG, "[screenshot]  | appid: {}", appID);
+
+    bool isInteractive = options.count("interactive") && options["interactive"].get<bool>();
+
+    // make screenshot
+    const std::string FILE_PATH = "/tmp/xdph_screenshot.png";
+    const std::string SNAP_CMD  = "grim " + FILE_PATH;
+    const std::string SNAP_INTERACTIVE_CMD = "grim -g \"$(slurp)\" " + FILE_PATH;
```

Comment:
I would, at launch, in init, check for these programs' existence. If they don't exist, don't create the portal.